File: ueueo-framework/ueueo-uow/src/main/java/com/ueueo/uow/IDatabaseApiContainer.cs
using System;
using JetBrains.Annotations;
using Volo.Abp.DependencyInjection;

namespace Volo.Abp.Uow;

public interface IDatabaseApiContainer : IServiceProviderAccessor
{
    [CanBeNull]
    IDatabaseApi FindDatabaseApi(@NonNull String key);

    void AddDatabaseApi(@NonNull String key, @NonNull IDatabaseApi api);

    [NotNull]
    IDatabaseApi GetOrAddDatabaseApi(@NonNull String key, @NonNull Func<IDatabaseApi> factory);
}

File: ueueo-frameworks/ueueo-uow/src/main/java/com/ueueo/uow/IDatabaseApiContainer.java
package com.ueueo.uow;

import org.springframework.lang.NonNull;
import org.springframework.lang.Nullable;

import java.util.function.Supplier;

/**
 * @author Lee
 * @date 2022-05-19 20:43
 */
public interface IDatabaseApiContainer {
    @Nullable
    IDatabaseApi findDatabaseApi(@NonNull String key);

    void addDatabaseApi(@NonNull String key, @NonNull IDatabaseApi api);

    @NonNull
    IDatabaseApi getOrAddDatabaseApi(@NonNull String key, @NonNull Supplier<IDatabaseApi> factory);
}
